### PR TITLE
Refactor gru_rewriter

### PR DESCRIPTION
**Pull request by lucienwang1009**
refactor gru rewriter and remove unused code. all tests are green.

### CODE REVIEW

**Comment by lucienwang1009** [comment]
@zhijxu-MS pls take a look at GRU rewriter change.

File: tf2onnx/rewriter/gru_rewriter.py [hunk 1]
```diff
@@ -78,146 +59,66 @@ def get_weight_and_bias(self, match):
...
-            enter_target_node = self.g.get_node_by_output(enter_target_node_input_id)
-            if is_tensor_array_op(enter_target_node):
-                log.debug("found output switch node")
-                return enter_target_node_input_id
-            log.debug("found enter target node is not ta node")
-            return None
-        log.debug("%d TensorArrayWriteV3 matching found, cannot validate output switch", len(ta_write_nodes))
+    def _state_variable_finder(self, context):
+        if self.gru_cell_type == RNNUnitType.GRUCell:
+            lstm_cell = context.cell_match
```
Comment (nbcsm) [inline tf2onnx/rewriter/gru_rewriter.py]: lstm?

**Comment by nbcsm** [comment]
we can put dropout check in another pr.

**Comment by lucienwang1009** [comment]
> we can put dropout check in another pr.

Sure.

**Pull request merged**